File: app/src/main/java/com/lostincontext/playlists/PlaylistsPresenterModule.java
package com.lostincontext.playlists;

import dagger.Module;
import dagger.Provides;

@Module
public class PlaylistsPresenterModule {

    private final PlaylistsContract.View view;

    public PlaylistsPresenterModule(PlaylistsContract.View view) {
        this.view = view;
    }

    @Provides
    PlaylistsContract.View provideMainScreenContractView() {
        return view;
    }

}

File: app/src/main/java/com/lostincontext/playlists/PlaylistsPresenterModule.kt
package com.lostincontext.playlists

import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class PlaylistsPresenterModule(private val view: PlaylistsContract.View,
                               private val userId: Long) {

    @Provides
    internal fun provideMainScreenContractView(): PlaylistsContract.View {
        return view
    }

    @Provides
    @Named("userId")
    internal fun provideUserId() : Long {
        return userId
    }

}
